### ISSUE TITLE
[Change Proposal] Allow orphaned transform destination index to be deleted

### ISSUE THREAD

**Issue by kcreddy** [elastic/package-spec#523]
While working on a feature for expiring indicators of compromise (IOC) of threat intel (TI) packages, we took an approach that involves `latest` transform. Essentially the transform ([example transform for `ti_recordedfuture`](https://github.com/elastic/integrations/pull/5460/files#diff-b770091f343a2afce6f2f253658bf92c46c99217dbb3861bfc8b53e34005efd2)) is going to create destination indices (`*_latest*`) which are used by the users to setup detection rules. The destination index itself is versioned with a version suffix such as `-1`, `-2`,.. so that we can support any breaking change made to the index mapping. 

Now the issue is this that when a destination index is eventually bumped up (lets say from `-1` to `-2`), the current behaviour doesn't delete the old destination index which is now orphaned as it has no links to the updated transform. This creates duplicates for readers using wildcards (`logs-*_latest-*`) or even in the false detection alerts when used with wildcards.

We need a mechanism to automatically purge the old destination index so that readers using wildcards (`logs-*_latest-*`) are not getting results from the old index.

**Comment by juliaElastic** [comment]
hey @jsoriano could you take a look at this to advise?

**Comment by jsoriano** [comment]
> Now the issue is this that when a destination index is eventually bumped up (lets say from `-1` to `-2`), the current behaviour doesn't delete the old destination index which is now orphaned as it has no links to the updated transform.

@kcreddy when does this bumping up happen? Does it happen when a new version of the package is installed?

**Comment by kcreddy** [comment]
> when does this bumping up happen? Does it happen when a new version of the package is installed?

@jsoriano, Yes it happens when new package version is installed. For example we have this PR for fixing the destination index mapping and we would like to increment the index version. 
https://github.com/elastic/integrations/pull/6514/files#diff-9c0b6d05505edf894e674f59e9af2d5721e621526b356cb29c8f8dbbbc603814

**Comment by jsoriano** [comment]
> > when does this bumping up happen? Does it happen when a new version of the package is installed?
> 
> @jsoriano, Yes it happens when new package version is installed.

I guess that then EPM could take care of the migration to the new version and the deletion of old indexes, as part of the package installation process. @juliaElastic wdyt?

**Comment by juliaElastic** [comment]
@jsoriano You mean the package installation logic in kibana, right? I agree, we could enhance that to delete old index. Can we transfer this issue to kibana repo then?
